### PR TITLE
Feature: global drag'n'drop

### PR DESCRIPTION
**Pull request by shamoon**
This PR was inspired by #245 , it creates a "global" drag-and-drop feature on any page other than the dashboard (where there's still the upload box, I thought it was wise to keep that since this feature isnt really obvious). Essentially it stays out of the way but for users who want to upload it can be done from anywhere, I dont really see any downsides, do others?

I also fixed the fact that toasts dont currently fade in / out. Fading is nice.

Video below.

https://user-images.githubusercontent.com/4887959/157983364-8a2e6e8b-3d95-4e48-98bb-ae31b181c81a.mov

### CODE REVIEW

**Comment by sukisoft** [comment]
Hm, could be a nice feature, but i think this is not really obvious to the user, that it exists. Since one has to indicate a drop at first, and after that the possibility to drop is displayed. Yet i dont see a chance of displaying it properly, since the view is already "full of buttons and/or documents". Typical presentation would be a globally visible panel or something like that, which says "drop files to upload" or something like that.

For me personally i would not use that feature, i can live with the upload chooser on the dashboard.

**Review by lippoliv — CHANGES_REQUESTED** [review]

File: src-ui/src/app/app.component.html [hunk 1]
```diff
@@ -1,3 +1,13 @@
 <app-toasts></app-toasts>
 
-<router-outlet></router-outlet>
+<ngx-file-drop dropZoneClassName="main-dropzone" contentClassName="main-content" [disabled]="!dragDropEnabled"
+(onFileDrop)="dropped($event)" (onFileOver)="fileOver()" (onFileLeave)="fileLeave()">
+    <ng-template ngx-file-drop-content-tmp>
+        <div class="global-dropzone-overlay fade" [class.show]="fileIsOver" [class.hide]="hidden">
+            <h2>Drop files to begin upload</h2>
```
[the remark this answers is not in the document]
Comment (shamoon) [reply src-ui/src/app/app.component.html]: Thank you, great catch! How English-centric of me =)

**Comment by shamoon** [comment]
Thanks, I fixed that and also fixed a random bug I found with the CSS spacing of the filter editor (literally it looks exactly the same just fixes a small overflow thing)

**Review by qcasey — APPROVED** [review]
Awesome, thank you! This is a very useful feature.

**Comment by shamoon** [comment]
Sorry 1 thing! for FrankStrieter =)

**Comment by shamoon** [comment]
Okey-dokey =)

**Pull request merged**